### PR TITLE
Api 5165 move date time parameter logic

### PR DESCRIPTION
**Pull request by eclendenning-va**
merge this before the dq pr

### CODE REVIEW

File: fhir-common/pom.xml [hunk 1]
```diff
@@ -9,5 +9,13 @@
   <modelVersion>4.0.0</modelVersion>
   <artifactId>fhir-common</artifactId>
   <version>6.0.4-SNAPSHOT</version>
+  <dependencies>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>28.2-jre</version>
```
Comment (jhulbert-va) [inline fhir-common/pom.xml:16]: `<properties/>`

File: fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameters.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    } else {
+      prefix = SearchPrefix.EQ;
+      date = paramString;
+    }
+  }
+
+  /**
+   * Indicates if the given date range (epoch millis) satisfies this date-time parameter.
+   *
+   * <p>See JpaDateTimeParameter.toQuerySnippet(...)
```
Comment (jhulbert-va) [inline fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameters.java:50]: Copy Pasta!

File: fhir-common/pom.xml [hunk 3]
```diff
@@ -9,5 +9,13 @@
   <modelVersion>4.0.0</modelVersion>
   <artifactId>fhir-common</artifactId>
   <version>6.0.4-SNAPSHOT</version>
+  <dependencies>
+    <dependency>
+      <groupId>com.google.guava</groupId>
```
Comment (jblefkove-va) [inline fhir-common/pom.xml:14]: Don't bring in guava just for the sake of `Preconditions.checkArgument`. That can be reimplemented locally.

File: fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameters.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Locale;
+import lombok.Data;
+
+@Data
+public class FhirDateTimeParameters implements Serializable {
```
Comment (jblefkove-va) [inline fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameters.java:16]: Should this be `FhirDateTimeParameter`, singular?

File: fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameters.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.Locale;
+import lombok.Data;
+
+@Data
```
Comment (jblefkove-va) [inline fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameters.java:15]: Can this still be `@Value`?

File: fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameters.java (hunk 4 above)
Comment (jblefkove-va) [inline fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameters.java:16]: I don't think this still needs to be `Serializable.` That was originally for JPA.

File: fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameters.java [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+  private static final int YEAR_MONTH = 7;
+
+  private static final int YEAR_MONTH_DAY = 10;
+
+  private static final int TIME_ZONE = 20;
+
+  private static final int TIME_ZONE_OFFSET = 25;
+
+  public SearchPrefix prefix;
```
Comment (jblefkove-va) [inline fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameters.java:27]: public?

File: pom.xml [hunk 7]
```diff
@@ -8,6 +8,7 @@
   </parent>
   <artifactId>fhir-resources-parent</artifactId>
   <version>6.0.4-SNAPSHOT</version>
+  <properties/>
```
Comment (bschofield-va) [inline pom.xml:11]: ?

File: fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameter.java [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+      case SA:
+        // and the range above the search value contains the range of the target value
+        return upperBound < lower;
+
+      case EB:
+        // and the range below the search value contains the range of the target value
+        return upper < lowerBound;
+
+      case AP:
+        throw new UnsupportedOperationException("AP search prefix not implemented");
```
Comment (bschofield-va) [inline fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameter.java:87]: Let's drop something in the backlog to add `ap` support.

**Review by bschofield-va — DISMISSED** [review]

**Review by bschofield-va — APPROVED** [review]

File: fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameter.java (hunk 8 above)
Comment (eclendenning-va) [reply fhir-common/src/main/java/gov/va/api/health/fhir/api/FhirDateTimeParameter.java:87]: https://vajira.max.gov/browse/API-5281

**Pull request merged**